### ISSUE TITLE
format-default: No coercion operator is defined

### ISSUE THREAD

**Issue by buchanantim** [PowerShell/PowerShell#14632]
## Steps to reproduce

```powershell
using namespace System.Collections.Generic

[LinkedList[Hashtable]]::new().AddLast( @{
    k1 = "v1"
})
```

## Expected behavior

When this is executed in Windows Powershell, the hashtable is added to the linked list, and the link list is sent to the output stream.
```none
List                           Next Previous Value
----                           ---- -------- -----
{System.Collections.Hashtable}               {k1} 
```

## Actual behavior

```none
format-default: No coercion operator is defined between types 'System.Collections.Hashtable&' and 'System.Object'.
```
## Environment data

```none

Name                           Value
----                           -----
PSVersion                      7.1.1
PSEdition                      Core
GitCommitId                    7.1.1
OS                             Microsoft Windows 10.0.19041
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0ΓÇª}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by SeidChr** [comment]
i have the very same problem. did you manage to find a workarround?
pwsh 7.1.3

i figured that it disappears when running in an "measure-command" scriptblock.

**Comment by vexx32** [comment]
Is the linked list handing out raw _pointers_ when enumerated? I guess that sort of makes sense... Yeah, guess the formatter just straight up doesn't expect that.

**Comment by SeidChr** [comment]
i just figured out that you can hide the error, and it will just work

![image](https://user-images.githubusercontent.com/1927076/121956892-eefcd180-cd61-11eb-9b86-42140871a298.png)

**Comment by iSazonov** [comment]
It is Formatting subsystem issue. I suppose it's not that hard to fix. (But we might have wondered about a generic support for linked lists in the language.)

**Comment by SeeminglyScience** [comment]
Similar to #8765 but for managed pointers (in this case a ref return property like `public ref Hashtable => ref _item;`)

**Comment by aolszowka** [comment]
I believe I am encountering this as well (if not I am happy to open a new issue):

## Steps to Reproduce
```powershell
[System.Collections.Generic.LinkedList[long]]$primeSieve = [System.Collections.Generic.LinkedList[long]]::new()
$primeSieve.Add(2)
$primeSieve.Last
```

Behavior seems to have changed since PowerShell 5.1

## PowerShell 7.1.2
```text
format-default: No coercion operator is defined between types 'System.Int64&' and 'System.Object'.
```

Environment Data
```text
Name                           Value
----                           -----
PSVersion                      7.1.2
PSEdition                      Core
GitCommitId                    7.1.2
OS                             Microsoft Windows 10.0.19042
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}       
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

## PowerShell 5.1
```text
List Next Previous Value
---- ---- -------- -----
{2}                    2
```

Environment Data
```text
Name                           Value
----                           -----
PSVersion                      5.1.19041.1023
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.19041.1023
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

## Workaround
In my case it was my bad, I really wanted to be doing `$primeSieve.Last.Value` anyway, which works in both versions of PowerShell, just weird to hit this and the difference between versions.

**Comment by rkeithhill** [comment]
I can repro this in both 7.1.3 and 7.2.0-preview.7.  Seems to be a regression.

**Comment by vexx32** [comment]
Looking at the docs, I can see that the `ValueRef` property seems to be new in .NET Core. So it's more something was added in .NET Core to this object type that the formatter doesn't know how to handle.

I still agree it should be fixed though; it seems like a bit of an edge case, but the formatter being entirely unable to render this kind of property is a little problematic I suppose.

**Comment by SeeminglyScience** [comment]
I would caution against treating this like an edge case.  It is currently, but this pattern is likely to pop up with increasing frequency.

Ideally PowerShell would know how to handle these properties as if they were any other indexer with both a getter and a setter.  The setter will be a lot more challenging since iirc `SLE.Expression` will throw with a method invocation on the LHS of `Assign` but for `get` it should be able to just dereference.

**Comment by bigboy-pdb-spam** [comment]
I got the error "format-default: No coercion operator is defined between types 'System.Int32&' and 'System.Object'." using `$l = [System.Collections.Generic.LinkedList[double]]::new(); $l.AddLast(1.3)` in the terminal. When I wrap the call in a `try` `catch` block (`try { $l.AddLast(1.3) } catch {}`) in the terminal it still generates the error. However, when I defined a class that extended `System.Collections.Generic.LinkedList[double]` and called `$this.AddLast` in a `try` `catch` block, within a script and executed it, the error wasn't thrown.
